pkg/scheduler/core/generic_scheduler.go

```diff
@@ -190,7 +190,20 @@ func (g *genericScheduler) assignReplicas(
 			return nil, fmt.Errorf("unsupported replica scheduling strategy, replicaSchedulingType: %s, replicaDivisionPreference: %s, "+
 				"please try another scheduling strategy", placement.ReplicaSchedulingType(), placement.ReplicaScheduling.ReplicaDivisionPreference)
 		}
-		return assignFunc(state)
+		assignResults, err := assignFunc(state)
+		if err != nil {
+			return nil, err
+		}
+		for i, removed, resultLen := 0, 0, len(assignResults); i < resultLen; i++ {
+			j := i - removed
+			// when the replicas of a scheduling result is zero,
+			// it should not be propagated to member cluster.
+			if assignResults[j].Replicas == 0 {
+				assignResults = append(assignResults[:j], assignResults[j+1:]...)
```

Comment:
It seems like we may move these items more than once if there are more than one empty cluster. Can we just loop the result and ignore the empty one?

```go
targetClusters := make([]workv1alpha2.TargetCluster, len(assignResults))
for i, cluster := range assignResults {
	if cluster.Replicas > 0 {
		targetClusters[i] = workv1alpha2.TargetCluster{Name: cluster.Name, cluster.Replicas}
	}
}
return targetClusters
```

Reply:
> It seems like we may move these items more than once if there are more than one empty cluster. 

Yes, so i use `j := i - removed` and `assignResults = append(assignResults[:j], assignResults[j+1:]...)` handle this situation。

> targetClusters := make([]workv1alpha2.TargetCluster, len(assignResults))


We cannot use this patch because the last few items will have the following situation.

```
targetClusters last item is 
{ Name:"", Replicas: 0}
```

Reply:
> ```
> targetClusters last item is 
> { Name:"", Replicas: 0}
> ```

Could you enlighten me in what kind of situation?

Reply:
> targetClusters := make([]workv1alpha2.TargetCluster, len(assignResults))


![image](https://github.com/karmada-io/karmada/assets/17962021/d97e9a9f-7ad7-4907-9a36-08ae0d21ccae)

Reply:
Oh sorry, my fault. I think the code is: 

```go
targetClusters := make([]workv1alpha2.TargetCluster, 0, len(assignResults))
for i, cluster := range assignResults {
	if cluster.Replicas > 0 {
		targetClusters[i] = workv1alpha2.TargetCluster{Name: cluster.Name, cluster.Replicas}
	}
}
return targetClusters
```
It looks easier.

Reply:
`cluster.Replicas` equal to 0 is a rare case, so my patch cannot execute `append` every time.

Reply:
Fair enough, but still got some disadvantages. Like this case, '0, 0, 0, 0, 0, 1'. It seems that each loop needs to move the rear items unless the last one.

Reply:
Updated. /cc @Garrybest